generic3g/actions/VerticalRegridAction.F90

```diff
@@ -87,36 +101,65 @@ subroutine update(this, importState, exportState, clock, rc)
       real(ESMF_KIND_R4), pointer :: x_in(:,:,:)
       real(ESMF_KIND_R4), pointer :: x_out(:,:,:)
 
-      real(ESMF_KIND_R4), pointer :: v_in(:,:,:)
-      real(ESMF_KIND_R4), pointer :: v_out(:,:,:)
+      real(ESMF_KIND_R4), pointer :: v_in(:)
+      real(ESMF_KIND_R4), pointer :: v_out(:)
 
-      integer :: i, j, k
-      integer, parameter :: IM = 2, JM = 2, LM = 2
+      integer :: istart, iend, jstart, jend, i, j
 
-      if (associated(this%v_in_coupler)) then
-         call this%v_in_coupler%run(phase_idx=GENERIC_COUPLER_UPDATE, _RC)
-      end if
+      ! if (associated(this%v_in_coupler)) then
+      !    call this%v_in_coupler%run(phase_idx=GENERIC_COUPLER_UPDATE, _RC)
+      ! end if
 
-      if (associated(this%v_out_coupler)) then
-         call this%v_out_coupler%run(phase_idx=GENERIC_COUPLER_UPDATE, _RC)
-      end if
+      ! if (associated(this%v_out_coupler)) then
+      !    call this%v_out_coupler%run(phase_idx=GENERIC_COUPLER_UPDATE, _RC)
+      ! end if
 
       call ESMF_StateGet(importState, itemName='import[1]', field=f_in, _RC)
-      call ESMF_StateGet(exportState, itemName='export[1]', field=f_out, _RC)
-
       call ESMF_FieldGet(f_in, fArrayPtr=x_in, _RC)
+
+      call ESMF_StateGet(exportState, itemName='export[1]', field=f_out, _RC)
       call ESMF_FieldGet(f_out, fArrayPtr=x_out, _RC)
 
-      call ESMF_FieldGet(this%v_in_coord, fArrayPtr=v_in, _RC)
-      call ESMF_FieldGet(this%v_out_coord, fArrayPtr=v_out, _RC)
+      istart = lbound(x_out, 1); iend = ubound(x_out, 1)
+      jstart = lbound(x_out, 2); jend = ubound(x_out, 2)
```

Comment:
This is hardcoded to a 2D horizontal representation.  Needs to use the feature implemented by @darianboggs to abstract a 3D pointer from any field.  

The existing call also fails if there are ungridded dims even if using a normal CS grid.

Reply:
Is this correct? https://github.com/GEOS-ESM/MAPL/blob/e7c8570b9e68368ad11b0190c013adf97e26497c/generic3g/actions/VerticalRegridAction.F90#L114-L126

Reply:
No - there is a higher level function that directly returns the pointer with the dimensions reshaped to what you need in your logic.  This will avoid the need to overload the FieldGet calls for different rank x_in and x_out.  Discuss with @darianboggs

Reply:
I think what you want is:

Delete Line 116

`   call assign_fptr_condensed_array(f_in, x_shape, x_in, _RC)` ! Line 117

and

Delete Line 121

`   call assign_fptr(f_in, x_shape, x_out, _RC)` ! Line 122

You may not need the `x_shape` after making these changes.

I'm happy to talk more about it. Just send a message when you want to talk.

Reply:
>    `call assign_fptr_condensed_array(f_in, x_shape, x_in, _RC) ! Line 117`

I had completely missed the `assign_fptr_condensed_array` routine. Fixed it now (hopefully)

Reply:
> > `call assign_fptr_condensed_array(f_in, x_shape, x_in, _RC) ! Line 117`
> 
> I had completely missed the `assign_fptr_condensed_array` routine. Fixed it now (hopefully)

https://github.com/GEOS-ESM/MAPL/blob/b133022959ddbd47d2f53dc2572bdfbbdc568f54/generic3g/actions/VerticalRegridAction.F90#L112-L116